packages/pool/contracts/StateUtils.sol

```diff
@@ -78,6 +85,11 @@ contract StateUtils is IStateUtils {
     /// @notice Epoch index of the most recent reward payment
     uint256 public epochIndexOfLastRewardPayment;
 
+    // Snapshot block number of the last vote created at one of the DAO
+    // Api3Voting apps
+    uint256 private lastVoteSnapshotBlock;
+    mapping(uint256 => uint256) private snapshotBlockToTimestamp;
```

Comment:
Need to also keep this because we want at most `MAX_INTERACTION_FREQUENCY` checkpoints created in the last `EPOCH_LENGTH` but checkpoints are accounted for in blocks, not timestamps.